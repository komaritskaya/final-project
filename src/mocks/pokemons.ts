import { Pokemon } from '../types';

const pokemons: Pokemon[] = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: true,
    catchDate: new Date(),
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
];

export default pokemons;
