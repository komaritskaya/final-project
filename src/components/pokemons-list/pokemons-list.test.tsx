import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import PokemonsList from './pokemons-list';
import pokemons from '../../mocks/pokemons';

const mockStore = configureStore([]);

it('Should Pokemons List component render correctly', () => {
  const store = mockStore({
    data: {
      pokemons,
    },
  });

  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <PokemonsList
            pokemons={pokemons}
          />
        </Router>
      </Provider>,
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
