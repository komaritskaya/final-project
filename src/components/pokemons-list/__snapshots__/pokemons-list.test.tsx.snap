// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Pokemons List component render correctly 1`] = `
<div>
  <div>
    <a
      href="/1"
      onClick={[Function]}
    >
      bulbasaur
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/2"
      onClick={[Function]}
    >
      ivysaur
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/3"
      onClick={[Function]}
    >
      venusaur
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/4"
      onClick={[Function]}
    >
      charmander
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/5"
      onClick={[Function]}
    >
      charmeleon
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/6"
      onClick={[Function]}
    >
      charizard
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/7"
      onClick={[Function]}
    >
      squirtle
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/8"
      onClick={[Function]}
    >
      wartortle
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/9"
      onClick={[Function]}
    >
      blastoise
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/10"
      onClick={[Function]}
    >
      caterpie
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/11"
      onClick={[Function]}
    >
      metapod
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/12"
      onClick={[Function]}
    >
      butterfree
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/13"
      onClick={[Function]}
    >
      weedle
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/14"
      onClick={[Function]}
    >
      kakuna
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/15"
      onClick={[Function]}
    >
      beedrill
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/16"
      onClick={[Function]}
    >
      pidgey
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/17"
      onClick={[Function]}
    >
      pidgeotto
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/18"
      onClick={[Function]}
    >
      pidgeot
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/19"
      onClick={[Function]}
    >
      rattata
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
  <div>
    <a
      href="/20"
      onClick={[Function]}
    >
      raticate
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      catch
    </button>
    <hr />
  </div>
</div>
`;
